rpc/src/v1/helpers/dispatch.rs

```diff
@@ -126,7 +126,10 @@ impl<C: miner::BlockChainClient, M: MinerService> FullDispatcher<C, M> {
 	pub fn dispatch_transaction(client: &C, miner: &M, signed_transaction: PendingTransaction) -> Result<H256> {
 		let hash = signed_transaction.transaction.hash();
 
-		miner.import_own_transaction(client, signed_transaction)
+		// use `import_claimed_local_transaction` so we can decide (based on config flags) if we want to treat
+		// it as local or not. Nodes with public RPC interfaces will want these transactions to be treated like
+		// external transactions.
+		miner.import_claimed_local_transaction(client, signed_transaction)
```

Comment:
@XertroV Can we only use `import_claimed_local_transaction` for `eth_sendRawTransaction`? I'm worried about a race condition where:

* A new transaction, from a vault account, is sent through `eth_sendTransaction`.
* Before the dispatcher gets a chance, the vault is closed, and thus the account is removed from address list.
* The dispatcher gives the transaction to miner. Miner doesn't find it in the account list, so it's imported as external transaction -- clearly this is not what we want.

A simple way may be just to add a boolean param in `dispatch_transaction`, and only set that to true for `eth_sendRawTransaction`.

Reply:
Yeah okay. I'm not too familiar with how all of this connects together, FYI.
What I'm doing atm is adding a `trusted: bool` param to `dispatch_transaction` that gets called with `false` from `eth_sendRawTx` and `true` from the other uses of `dispatch_transaction` (which seems to be called from `personal.rs` or `signer.rs`. That should mean all RPC methods that involve signing get a free pass regardless of config options.

Also going to add this flag to `import_claimed_local_transaction`.